graylog2-web-interface/src/components/pipelines/PipelineListItem.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+import { isPermitted } from 'util/PermissionsMixin';
+import { Link, LinkContainer } from 'components/common/router';
+import Routes from 'routing/Routes';
+import { CounterRate, MetricContainer } from 'components/metrics';
+import PipelineConnectionsList from 'components/pipelines/PipelineConnectionsList';
+import { Button, Label } from 'components/bootstrap';
+import type { PipelineType } from 'stores/pipelines/PipelinesStore';
+import type { PipelineConnectionsType } from 'stores/pipelines/PipelineConnectionsStore';
+import type { Stream } from 'logic/streams/types';
+import { defaultCompare as naturalSort } from 'logic/DefaultCompare';
+import useCurrentUser from 'hooks/useCurrentUser';
+import useGetPermissionsByScope from 'hooks/useScopePermissions';
+
+import ButtonToolbar from '../bootstrap/ButtonToolbar';
+import { Spinner } from '../common';
+
+type Props = {
+  pipeline: PipelineType;
+  pipelines: Array<PipelineType>;
+  connections: Array<PipelineConnectionsType>;
+  streams: Array<Stream>;
+  onDeletePipeline: () => void;
+};
+const PipelineNameTD = styled.td`
+  max-width: 300px;
+  overflow-x: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  width: 300px;
+`;
+const StreamListTD = styled.td`
+  max-width: 150px;
+  width: 150px;
+  word-wrap: break-word;
+`;
+
+const PipelineStage = styled.div<{ $idle?: boolean }>(
+  ({ $idle, theme }) => css`
+    border: 1px solid ${theme.colors.gray[$idle ? '50px' : '70px']};
+    border-radius: 4px;
+    display: inline-block;
+    margin-right: 15px;
+    padding: 20px;
+    text-align: center;
+    width: 120px;
+    background-color: ${$idle
+      ? theme.utils.colorLevel(theme.colors.global.contentBackground, 10)
+      : theme.colors.global.contentBackground};
+  `,
+);
+const DefaultLabel = styled(Label)`
+  display: inline-flex;
+  margin-left: 5px;
```

Comment:
There are a lot of hard coded px values in these styled components maybe for some of them we can use theme spacing?